### PR TITLE
doc: install pyocd using pip2

### LINKED ISSUE #5699: Zephyr installs a broken pyOCD
The Zephyr file `scripts/requirements.txt` pulls in pyOCD:

https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/requirements.txt

That file is used to install Python 3 dependencies using pip(3) in Zephyr's getting started documentation:

http://docs.zephyrproject.org/getting_started/installation_linux.html#installing-requirements-and-dependencies
http://docs.zephyrproject.org/getting_started/installation_mac.html#installing-requirements-and-dependencies
http://docs.zephyrproject.org/getting_started/installation_win.html#installing-requirements-and-dependencies

pyOCD doesn't work on Python 3, though, as described in this issue:

https://github.com/mbedmicro/pyOCD/issues/208

There are two changes that are needed to fix pyOCD use on Zephyr until that issue is resolved.

1. Zephyr must remove the pyOCD line from scripts/requirements.txt
2. The Zephyr documentation must clarify that if pyOCD is being used to flash a board (any board), it must be installed with Python 2

A fix for this issue requires checking around all the individual board documentation pages in addition to the getting started / installation instructions.

### PR DESCRIPTION
**Pull request by nashif**
pyocd does not exist as a python 3 module, use pip2 to install it.

Fixes #5699

### CODE REVIEW

**Comment by galak** [comment]
Should we just give the command line to install pyOCD without requirements-2 file?

**Comment by nashif** [comment]
> Should we just give the command line to install pyOCD without requirements-2 file?

we could, but based on experience, some other module might be needed later which will lead to us being out of sync with the packages that need to be installed. better to have them in one place.

File: doc/getting_started/installation_linux.rst [hunk 1]
```diff
@@ -70,6 +70,10 @@ Install additional packages required for development with Zephyr::
    cd ~/zephyr  # or to your directory where zephyr is cloned
    pip3 install --user -r scripts/requirements.txt
 
+To install pyocd, which is a python2 module, please use::
```
Comment (dbkinder) [inline doc/getting_started/installation_linux.rst]: how about:

    If you require pyocd, (an open source python2 library for programming and
    debugging ARM Cortex-M microcontrollers), use this command::

**Review by galak — DISMISSED** [review]
How about py2-requirements.txt, requirements-2.txt doesn’t seem clear.

File: doc/getting_started/installation_mac.rst [hunk 2]
```diff
@@ -64,6 +64,10 @@ Install tools to build Zephyr binaries:
    cd ~/zephyr   # or to the folder where you cloned the zephyr repo
    pip3 install --user -r scripts/requirements.txt
 
+To install pyocd, a python 2 only  module, use::
```
Comment (dbkinder) [inline doc/getting_started/installation_mac.rst]: Same here:

    If you require pyocd, an open source python2 library for programming and
    debugging ARM Cortex-M microcontrollers, use this command::

File: doc/getting_started/installation_win.rst [hunk 3]
```diff
@@ -88,6 +88,11 @@ packages from their respective websites.
       cd %userprofile%\zephyr
       pip install --user -r scripts/requirements.txt
 
+#. If you need to install pyocd, it is available as a python 2 module only, this
```
Comment (dbkinder) [inline doc/getting_started/installation_win.rst]: And here:

    #. If you require pyocd, an open source python2 library for programming and
       debugging ARM Cortex-M microcontrollers, use this command::

**Review by dbkinder — DISMISSED** [review]
might be good to explain only ARM developers might need the pyocd module...

**Review by tyrantlizard77 — APPROVED** [review]
That's unfortunate there's no python3 build.
Wonder if we can file a bug somewhere on it.

**Review by dbkinder — APPROVED** [review]

**Comment by dbkinder** [comment]
https://github.com/mbedmicro/pyOCD/issues/208

**Pull request merged**